src/tools/vector2d.hpp

```diff
@@ -152,7 +152,8 @@ template<typename Type, typename Alloc = std::allocator<Type>> class vector2d {
 		size_t old_w = w, old_h = h;
 		w = width; h = height;
 		data.resize(w * h);
-		if(old_w < width) {
+		// ASAN undefined behaviour if old_h==0, y=old_h-1 is ...
+		if(old_w < width && old_h) {
```

Comment:
I've rewritten this function with unit tests, so presumably this change doesn't apply anymore?